### PR TITLE
Movement Sync - 19271 Fix

### PR DESCRIPTION
**Pull request by Langerz82**
**Changes proposed:**

-  Fixes Movement Syncing. 
-  https://github.com/TrinityCore/TrinityCore/issues/19271

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #  (insert issue tracker number)
19271
possibly 19156

**Tests performed:** (Does it build, tested in-game, etc.)
Builds yes, tested in-game, havent inspected the packets to make sure solution is optimal.

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Inspect packets to make sure no flooding is going on.
- [ ] Possibly needs restruct code, needs to be evaluated by someone more knowledgeable.

### CODE REVIEW

**Comment by Langerz82** [comment]
Travis makes me sad. :(

**Comment by Jildor** [comment]
You forgot add PlayerMove.SendInitialMovementHeartbeat in worldserver.conf.dist

**Comment by Seyden** [comment]
@ccrs probably your special field

**Comment by Ovahlord** [comment]
I think you dont need to iterate through all players within range. SendDirectMessage does that already so just build the packet and let the player send it.

**Comment by Seyden** [comment]
SendDirectMessage only sends the packet to the specific player. What you mean is probably SendMessageToSet.. but i don't know.. SendMessageToSet does send it to all players in GetMap()->GetVisibilityDistance().. i don't know if there is a difference between the distance used in clientGUIDs and GetVisibilityDistance()

**Comment by illfated** [comment]
Any TC member who can restart the travis-ci build?
Looks like Travis has had a lot of failed builds lately.

**Comment by P-Kito** [comment]
I'm confused. 
Wasn't "Heartbeat" Mechanic the thing that makes CC (Like Hex, Fear) break early after a few "heartbeats"?

Sorry for the possible off topic 👺

File: src/server/game/Handlers/MovementHandler.cpp [hunk 1]
```diff
@@ -373,6 +373,21 @@ void WorldSession::HandleMovementOpcodes(WorldPacket& recvData)
...
     mover->m_movementInfo = movementInfo;
 
+    // Send Multiple Movement Messages if there is a new player visible.
+    if (sWorld->getBoolConfig(CONFIG_SEND_INITIAL_MOVEMENT_HEARTBEAT) && _player->_movementUpdate)
+    {
+        for (GuidUnorderedSet::iterator it = _player->m_clientGUIDs.begin(); it != _player->m_clientGUIDs.end(); ++it)
+        {
+            if (Player* target = ObjectAccessor::GetPlayer(_player->GetMap(), *it))
+            {
+                WorldPacket data(MSG_MOVE_HEARTBEAT, recvData.size());
```
[the remark this answers is not in the document]
Comment (Ovahlord) [reply src/server/game/Handlers/MovementHandler.cpp]: 3.3.5 does not use this standard yet. They have the classes for it but they dont use it yet

[the remark this answers is not in the document]
Comment (Seyden) [reply src/server/game/Handlers/MovementHandler.cpp]: Ah shit :D thx for noticing me

**Comment by Seyden** [comment]
Btw.. Master don't have MSG_MOVE_HEARTBEAT.. so the patch won't be valid for master

File: src/server/game/Handlers/MovementHandler.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ccrs) [reply src/server/game/Handlers/MovementHandler.cpp]: wrongggg, there are like two definitions... at least there ARE LUL

https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Server/Packets/SpellPackets.h

**Comment by ccrs** [comment]
btw, dont know if this is correct, have you checked sniffs? @chaodhib is the expert

**Comment by Shauren** [comment]
This is completely broken and doesn't work beyond 2 units/players

**Pull request closed without merging**